2022-11-27.rb

```diff
@@ -0,0 +1,61 @@
+require File.expand_path(File.dirname(__FILE__) + '/assert')
+require 'pry'
+require 'json'
+
+# Level: Medium
+# This problem was asked by Google.
+
+# Given the root to a binary tree, implement serialize(root), which serializes the tree into a string,
+# and deserialize(s), which deserializes the string back into the tree.
+# 
+# For example, given the following Node class:
+# 
+# class Node:
+#     def __init__(self, val, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+#
+# The following test should pass:
+# 
+# node = Node('root', Node('left', Node('left.left')), Node('right'))
+# assert deserialize(serialize(node)).left.left.val == 'left.left'
+
+# my Ruby implementation of the Node class given in the instructions in Python
+class Node
+  attr_reader :val, :left, :right
+
+  def initialize(val, left=nil, right=nil)
+    @val = val
+    @left = left
+    @right = right
+  end
+end
```

Comment:
```suggestion
Node = Struct.new(:val, :left, :right)
```

Reply:
@byroot that's great, I wasn't aware of this convenient way to bundle a number of attributes together! Thanks!

Reply:
I would even add `keyword_argument: true`
```
Node = Struct.new(:val, :left, :right, keyword_argument: true)
```
So you can do
```
 Node.new(val: "value", right: right, left: another_node)
 ```

https://www.bigbinary.com/blog/ruby-2-5-allows-creating-structs-with-keyword-arguments